opennlp-tools/src/test/java/opennlp/tools/eval/AbstractEvalTest.java

```diff
@@ -18,73 +18,118 @@
 package opennlp.tools.eval;
 
 import java.io.File;
-import java.io.IOException;
+import java.io.FileNotFoundException;
 import java.io.InputStream;
 import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import org.junit.Assert;
 
 import opennlp.tools.ml.maxent.quasinewton.QNTrainer;
 import opennlp.tools.ml.naivebayes.NaiveBayesTrainer;
 import opennlp.tools.ml.perceptron.PerceptronTrainer;
+import opennlp.tools.util.ObjectStream;
+import opennlp.tools.util.StringUtil;
 import opennlp.tools.util.TrainingParameters;
 import opennlp.tools.util.model.ModelUtil;
 
-public class EvalUtil {
+public abstract class AbstractEvalTest {
 
-  static final double ACCURACY_DELTA = 0.0001d;
+  public static final double ACCURACY_DELTA = 0.0001d;
+  public static final String HASH_ALGORITHM = "MD5";
 
-  static TrainingParameters createPerceptronParams() {
+  public static void verifyTrainingData(ObjectStream<?> samples, BigInteger checksum) throws Exception {
+
+    MessageDigest digest = MessageDigest.getInstance(HASH_ALGORITHM);
+
+    Object sample;
+    while ((sample = samples.read()) != null) {
+      digest.update(sample.toString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    samples.close();
+
+    Assert.assertEquals(checksum, new BigInteger(1, digest.digest()));
+
+  }
+  
+  public static void verifyFileChecksum(Path file, BigInteger checksum) throws Exception {
+
+    MessageDigest digest = MessageDigest.getInstance(HASH_ALGORITHM);
+
+    try (InputStream in = Files.newInputStream(file)) {
+      byte[] buf = new byte[65536];
+      int len;
+      while ((len = in.read(buf)) > 0) {
+        digest.update(buf, 0, len);
+      }
+    }
+
+    Assert.assertEquals(checksum, new BigInteger(1, digest.digest()));
+  }
+  
+  public static void verifyDirectoryChecksum(Path path, String extension, BigInteger checksum)
+      throws Exception {
+
+    MessageDigest digest = MessageDigest.getInstance(HASH_ALGORITHM);
+    
+    final List<Path> paths = Files.walk(path)
```

Comment:
Files.walk doesn't make any promised about the ordering of the files you see. Therefore you might end up computing checksums in this order a, b, c, d or maybe in this order b, a, d, c. Even when the checksums of the individual files are identical you would end up with a different overall checksum.